File: go/pkg/naming.go
package pkg

import "strings"

var namePatterns = []string{
    "%s%s",
    "%s%s",
    "%s-%s",
    "%s-%s",
    "%shub",
    "%slab",
}

func SuggestNames(profile TraitProfile, language string) []string {
    var names []string
    vibe := capitalize(profile.Vibe)
    theme := capitalize(profile.Theme)
    for _, pattern := range namePatterns {
        var name string
        switch pattern {
        case "%s%s", "%s-%s":
            name = formatPattern(pattern, vibe, theme)
        case "%shub", "%slab":
            name = formatPattern(pattern, vibe+theme, "")
        }
        names = append(names, Slugify(name, language))
    }
    return names
}

func formatPattern(pattern, a, b string) string {
    if strings.Contains(pattern, "%s-%s") || strings.Contains(pattern, "%s%s") {
        return strings.Replace(pattern, "%s%s", a+b, 1)
    }
    return strings.Replace(pattern, "%s", a, 1)
}

func capitalize(s string) string {
    if len(s) == 0 {
        return s
    }
    return strings.ToUpper(s[:1]) + s[1:]
}

File: nodejs/src/naming.js
const { slugify } = require('./utils');

const NAME_PATTERNS = [
  '{vibe}{theme}',
  '{theme}{vibe}',
  '{vibe}-{theme}',
  '{theme}-{vibe}',
  '{vibe}{theme}hub',
  '{theme}{vibe}lab',
];

function suggestNames(profile, language) {
  return NAME_PATTERNS.map((pattern) => {
    let name = pattern
      .replace('{vibe}', capitalize(profile.vibe))
      .replace('{theme}', capitalize(profile.theme));
    return slugify(name, language);
  });
}

function capitalize(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
}

module.exports = { suggestNames };